cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -715,7 +726,11 @@ private void doAddSchedule(HttpRequest request, HttpResponder responder, String
     List<Constraint> constraints = Objects.firstNonNull(scheduleFromRequest.getConstraints(), NO_CONSTRAINTS);
     ProgramSchedule schedule = new ProgramSchedule(scheduleName, description, programId, properties,
                                                    scheduleFromRequest.getTrigger(), constraints);
-    programScheduler.addSchedule(schedule);
+    try {
+      programScheduler.addSchedule(schedule);
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException(e);
```

Comment:
See https://issues.cask.co/browse/CDAP-8498 and the PR linked in the comments for an example where an `IllegalArgumentException` was unintentionally being translated into a `BadRequestException`.